cscs-checks/libraries/math/trilinos_compile_run.py

```diff
@@ -49,3 +51,13 @@ def __init__(self, linkage):
     def set_cxxflags(self):
         flags = self.prgenv_flags[self.current_environ.name]
         self.build_system.cxxflags = flags
+
+    @rfm.run_before('compile')
+    def set_prebuild_cmd(self):
+        if self.current_environ.name.startswith('PrgEnv-intel'):
+            if '20.06' in os.getenv('MODULERCFILE', ''):
```

Comment:
```suggestion
        if self.current_environ.name == 'PrgEnv-intel' and cray_cdt_version() ==  '20.06':
```

Where the `cray_cdt_version()` should be more elaborate than this check. `MODULERCFILE` is defined only when a `cdt` module is loaded; it doesn't cover the system default cdt. I think for that you should check the version from the following:

```
cat /opt/cray/pe/cdt/default/modulerc | head -1
```

Note: you should do that Pythonically, not by launching a new shell.

@victorusu Do you agree with the `cray_cdt_version()` function?

Reply:
I think also that we should move that function to a utility package. We are going to need it in different cases.

Reply:
I'm writing a utility function. I will do a PR soon.

Reply:
You can use this utility for retrieving the currently loaded CDT module: https://github.com/eth-cscs/reframe/pull/1416

Reply:
@ChristopherBignamini #1416 is now merged, so you can proceed with this PR.

Reply:
@vkarak, yes! The `cray_cdt_version` is a much better approach!